pandas/core/generic.py

```diff
@@ -4522,6 +4530,102 @@ def copy(self, deep=True):
         Returns
         -------
         copy : type of caller
+
+        Examples
+        --------
+        >>> s = pd.Series([1,2], dtype="int32", index=["a", "b"])
```

Comment:
Is the `dtype` argument provided because you want it to be clear that type is copied over and not automatically upcast to int64? I think that's what you are going for but it wasn't immediately obvious to me, so I'd try and make it more explicit if that is your intention